web/web_test.go

```diff
@@ -188,6 +191,11 @@ func TestPublicFilesRequest(t *testing.T) {
 	res := httptest.NewRecorder()
 	th.Web.MainRouter.ServeHTTP(res, req)
 	assert.Equal(t, helloHTML, res.Body.String())
+
+	req, _ = http.NewRequest("GET", "/plugins/com.mattermost.sample/nefarious-file-access.html", nil)
```

Comment:
Perhaps I don't follow the code, but I think this is just another 404 test as currently written. Propose writing as:
```suggestion
	req, _ = http.NewRequest("GET", "/plugins/com.mattermost.sample/public/../nefarious-file-access.html", nil)
```

to prove the security restriction.

Reply:
@lieut-data ,
I see what you are trying to do now. My code never actually sees the ../. There is something else in the system that sees the ../ and 301 redirects the requester to the URL I have above. That is why I am testing the redirected result to make sure it cannot be requested after the redirect. I assume the original redirect is in the http handler framework before it ever gets to my code. 

I can dig further if you wish. Maybe "../" in the URL should be handled elsewhere? Is that part of this story?

Reply:
Digging a bit deeper, it does indeed look like `net/http` cleans the path for us (and presumably our gorilla mug does the same). I think it's still worthwhile updating the test to explicitly assert the above. Fine to keep both!